components/ibm-components/watson/train/src/wml-train.py

```diff
@@ -68,73 +71,117 @@ def train(args):
                        "instance_id": wml_instance_id,
                        "apikey": wml_apikey
                       }
-    client = WatsonMachineLearningAPIClient( wml_credentials )
-
+    client = WatsonMachineLearningAPIClient ( wml_credentials )
     # define the model
-    metadata = {
-        client.repository.DefinitionMetaNames.NAME              : wml_run_definition,
-        client.repository.DefinitionMetaNames.AUTHOR_NAME       : wml_author_name,
-        client.repository.DefinitionMetaNames.FRAMEWORK_NAME    : wml_framework_name,
-        client.repository.DefinitionMetaNames.FRAMEWORK_VERSION : wml_framework_version,
-        client.repository.DefinitionMetaNames.RUNTIME_NAME      : wml_runtime_name,
-        client.repository.DefinitionMetaNames.RUNTIME_VERSION   : wml_runtime_version,
-        client.repository.DefinitionMetaNames.EXECUTION_COMMAND : wml_execution_command
+    lib_meta = {
+        client.runtimes.LibraryMetaNames.NAME: wml_run_definition,
+        client.runtimes.LibraryMetaNames.VERSION: wml_framework_version,
+        client.runtimes.LibraryMetaNames.FILEPATH: model_code,
+        client.runtimes.LibraryMetaNames.PLATFORM: {"name": wml_framework_name, "versions": [wml_framework_version]}
+    }
+    custom_library_details = client.runtimes.store_library(lib_meta)
+    custom_library_uid = client.runtimes.get_library_uid(custom_library_details)
+
+    # create a pipeline with the model definitions included
+    doc = {
+        "doc_type": "pipeline",
+        "version": "2.0",
+        "primary_pipeline": wml_framework_name,
+        "pipelines": [{
+            "id": wml_framework_name,
+            "runtime_ref": "hybrid",
+            "nodes": [{
+                "id": "training",
+                "type": "model_node",
+                "op": "dl_train",
+                "runtime_ref": wml_run_name,
+                "inputs": [],
+                "outputs": [],
+                "parameters": {
+                    "name": "tf-mnist",
+                    "description": wml_run_definition,
+                    "command": wml_execution_command,
+                    "training_lib_href": "/v4/libraries/"+custom_library_uid,
+                    "compute": {
+                        "name": "k80",
+                        "nodes": 1
```

Comment:
Maybe add new parameters for the name and node so users can change their t-shirt size?

Reply:
done